client/client.go

```diff
@@ -448,125 +384,42 @@ func (c *CfClient) setAnalyticsServiceClient(ctx context.Context) {
 // BoolVariation returns the value of a boolean feature flag for a given target.
 //
 // Returns defaultValue if there is an error or if the flag doesn't exist
-func (c *CfClient) BoolVariation(key string, target *evaluation.Target, defaultValue bool) (bool, error) {
-	fc := c.getFlagFromCache(key)
-	if fc != nil {
-		// load segments dep
-		c.getSegmentsFromCache(fc)
-
-		result := checkPreRequisite(c, fc, target)
-		if !result {
-			return fc.Variations.FindByIdentifier(fc.OffVariation).Bool(defaultValue), nil
-		}
-		variation, err := fc.BoolVariation(target)
-		if err != nil {
-			return defaultValue, err
-		}
-		c.analyticsService.PushToQueue(target, fc, variation)
-
-		return variation.Bool(defaultValue), nil
-	}
-	return defaultValue, nil
+func (c *CfClient) BoolVariation(key string, target *eval.Target, defaultValue bool) (bool, error) {
+	value := c.evaluator.BoolVariation(key, target, defaultValue, c.analyticsService.PushToQueue)
+	return value, nil
```

Comment:
This suggests we don't ever propagate errors up from a evaluation anymore.   Is it true that the evaluator will never error?

Reply:
I'm wondering, what happens here, if I pass in an empty string for key?

Reply:
This is for backward comp I cant change method signature now, if you put empty key it will serve default value

Reply:
I don't think we should change the method signature, just curious if there is any scenario where we would want to alert the user that the something went wrong, so we will be returning the defaultValue.   If we returned an error then the user could look into what happened.